builtin/mainmenu/dlg_create_world.lua

```diff
@@ -51,46 +154,172 @@ local function create_world_formspec(dialogdata)
 				end
 			end
 		end
+
+		local ds = (gameconfig:get("disallowed_settings") or ""):split()
+		for _, value in pairs(ds) do
+			disallowed_settings[value:trim()] = true
+		end
 	end
 
 	local mglist = ""
-	local selindex = 1
+	local selindex
 	local i = 1
+	local first_mg
 	for k,v in pairs(mapgens) do
+		if not first_mg then
+			first_mg = v
+		end
 		if current_mg == v then
 			selindex = i
 		end
 		i = i + 1
 		mglist = mglist .. v .. ","
 	end
+	if not selindex then
+		selindex = 1
+		current_mg = first_mg
+	end
 	mglist = mglist:sub(1, -2)
 
+	local mg_main_flags = function(mapgen, y)
+		if mapgen == "singlenode" then
+			return "", y
+		end
+		if disallowed_settings["mg_flags"] then
+			return "", y
+		end
+
+		local form = "checkbox[4.2," .. y .. ";flag_mg_caves;" ..
+			fgettext("Caves") .. ";"..chflag(flags.main, "caves").."]"
+		y = y + 0.5
+		form = form .. "checkbox[4.2,"..y..";flag_mg_dungeons;" ..
+			fgettext("Dungeons") .. ";"..chflag(flags.main, "dungeons").."]"
+		y = y + 0.5
+
+		if mapgen == "v6" then
+			form = form .. "checkbox[4.2,"..y..";flag_mg_decorations;" ..
+			fgettext("Decorations (without trees and jungle grass)") .. ";"
+			.. chflag(flags.main, "decorations").."]" ..
+			"tooltip[flag_mg_decorations;" ..
+			fgettext("Decorations are special structures provided by the game that appear on the surface. Note: This does not affect trees and jungle grass generated by the v6 mapgen")
+			.. "]"
+
+			y = y + 0.5
+		else
+			form = form .. "checkbox[4.2,"..y..";flag_mg_decorations;" ..
+			fgettext("Decorations") .. ";" ..
+			chflag(flags.main, "decorations").."]" ..
+			"tooltip[flag_mg_decorations;" ..
+			fgettext("Decorations are special structures that generate on the surface includes things such as plants, trees and other small extras")
+			.. "]"
+
+			y = y + 0.5
+		end
+		form = form .. "tooltip[flag_mg_caves;" ..
+		fgettext("Carve a large, interconnected network of tunnels and caves in the terrain")
```

Comment:
Unnecessary tooltip, people know what caves are.

Reply:
Or alternatively i mean: Why have an over-explaining tooltip for caves but nothing for dungeons?
If dungeons are obvious and need no tooltip, then caves are too, i do not understand the inconsistency.

Reply:
Nevermind, sorry, answered in thread:
"It's basically there to distinguish caves from caverns, two flags which are subtly different."
FIne for me.